### PR TITLE
Update prompt_sorin_setup (python support)

### PR DESCRIPTION
**Pull request by jeeftor**
Added python support

Please be sure to check out our [contributing guidelines](https://github.com/sorin-ionescu/prezto/blob/master/CONTRIBUTING.md)
before submitting your pull request.

Fixes #
Deals with half of issue #924

## Proposed Changes

  - Add `python-info` into the `precmd` (does not require async because is looking at env vars 
  - Made the virtual env color orange
  
sorin2

### CODE REVIEW

**Comment by belak** [comment]
Please avoid creating a new PR every time a change is made, though I do appreciate the additional information. You can push commits to your branch and they will update the PR without needing you to make a new one.

Anyway. this is a nice small change... though if there isn't a virtualenv set, this will add an extra space at the beginning of the rprompt, which changes how many characters are allowed before the rprompt disappears. Would it be possible to remedy that?

**Comment by jeeftor** [comment]
So maybe i'm doing something wrong but I don't actually see the `()`'s when i deactivate

sorin3

Are you sure its an issue?  In a previous version I had a function that would handle what you are talking about

```
# Returns the Virtual Environment string if it exists
function check_python_virtual_environment() {
  if [[ -z "${VIRTUAL_ENV// }" ]]; then
    current_python_virtual_environment=
  else
    current_python_virtual_environment="(${VIRTUAL_ENV[(ws:/:)-1]})"
  fi
}
```

but I wasn't actually using `zstyle` and now that i switched i don't see that behavior any more.

**Comment by jeeftor** [comment]
The only bug I do see is maybe i need to nix the space before `${editor_info` 
```
RPROMPT='$python_info[virtualenv] ${editor_info[overwrite]}%(?:: %F{1}⏎%f)${VIM:+" %B%F{6}V%f%b"}${_prompt_devel_git}'
```

I'll make that update - because in the case there is no pyenv i think $python_info is unset but i'm still adding a space after it so thats bad.

**Comment by jeeftor** [comment]
So to summarize things thus far

* When a virtual env is set prompt will get `(virtualenv) ` prepended to the `RPROMPT`
* When the virtual env is not set the prompt will have nothing prepended
* The `() ` is set inside a `zstyle` command so there will be no empty `()`'s in the case of no prompt (as far as i know)

File: modules/prompt/functions/prompt_sorin_setup [hunk 1]
```diff
@@ -133,7 +143,7 @@ function prompt_sorin_setup {
 
   # Define prompts.
   PROMPT='${SSH_TTY:+"%F{9}%n%f%F{7}@%f%F{3}%m%f "}%F{4}${_prompt_sorin_pwd}%(!. %B%F{1}#%f%b.)${editor_info[keymap]} '
-  RPROMPT='${editor_info[overwrite]}%(?:: %F{1}⏎%f)${VIM:+" %B%F{6}V%f%b"}${_prompt_sorin_git}'
+  RPROMPT='$python_info[virtualenv]${editor_info[overwrite]}%(?:: %F{1}⏎%f)${VIM:+" %B%F{6}V%f%b"}${_prompt_devel_git}'
```
Comment (indrajitr) [inline modules/prompt/functions/prompt_sorin_setup]: Where is `_prompt_devel_git` coming from? I am guessing that's a typo.

File: modules/prompt/functions/prompt_sorin_setup [hunk 2]
```diff
@@ -123,6 +129,10 @@ function prompt_sorin_setup {
   zstyle ':prezto:module:git:info:keys' format \
     'status' '%b %p %c:%s%A%B%S%a%d%m%r%U%u'
 
+
+  # Set python-info format
+  zstyle ':prezto:module:python:info:virtualenv' format '%f%F{208}(%v)%F{7} '
```
Comment (indrajitr) [inline modules/prompt/functions/prompt_sorin_setup]: We have to be careful with the colors that we use (particularly for default theme). Best to stick to the 16 terminal colors for maximum compatibility. We might have to have ellipsis for very long string (again, as safety for default theme).

**Comment by jeeftor** [comment]
@indrajitr  I made the two changes you asked for.  Good finds.  Also moving forward I'm happy to work on a generalized env setup for the prompt.  If the python changes are acceptable - at least i'm hoping we can push those.  

(I actually wont be able to test the changes until tomorrow - because i'm limited to web browser access only - but I think from looking over it - it should work)

File: modules/prompt/functions/prompt_sorin_setup (hunk 2 above)
Comment (jeeftor) [inline modules/prompt/functions/prompt_sorin_setup]: Updated in last commit

File: modules/prompt/functions/prompt_sorin_setup (hunk 1 above)
Comment (jeeftor) [inline modules/prompt/functions/prompt_sorin_setup]: This was a type-o on my end. Fixed in last commit.

**Comment by johnpneumann** [comment]
Would you mind taking one of the scripts (docker or vagrant, doesn't matter to me) from this comment https://github.com/sorin-ionescu/prezto/issues/1415#issuecomment-320886220 and post screen shots of some basic interactions from whichever one you choose. Some things I'd like to see:

- if/then/else
- nested if/then/else
- for/do/done
- nested for/do/done
- history search
- with python installed
- with python installed and a virtualenv activated
- without python installed

We've had a few issues with themes over the last few months, so I want to make sure that we aren't breaking existing functionality. Thanks for the help with this. The code looks good to me based on my quick glance through. Can dive deeper, but it looks like @indrajitr has already given some feedback, so I'll let him roll with it.

**Comment by indrajitr** [comment]
Didn't go all the way into trying in a dockerized instance. But looks good so far.
Further enhancements can happen as followup.

Thanks @jeeftor for your first PR! ✨

**Pull request merged**